nailgun/entity_mixins.py

```diff
@@ -378,7 +378,8 @@ def __init__(self, server_config=None, **kwargs):
         # Check that a valid set of field values has been passed in.
         if not set(kwargs.keys()).issubset(self._fields.keys()):
             raise NoSuchFieldError(
-                f'Valid fields are {self._fields.keys()}, but received {kwargs.keys()} instead.'
+                f'Valid fields are {list(self._fields.keys())}, invalid fields(s) passed '
+                f'to entity: {set(self._fields.keys()).difference(set(kwargs.keys()))}'
```

Comment:
I have the set difference backwards, good catch, but I wanted to improve the exception message here by actually informing the caller of what invalid keys were passed - with the current error message, it just lists all the keys and you have to go lookup the entity class to find out which one is invalid.

Current exception in master looks like:

![image](https://user-images.githubusercontent.com/21315076/102787065-80b8f280-436e-11eb-96ff-ffe22e799858.png)


I'm going to update my branch commit with the `set`s switched, so that the error message looks like this:
![image](https://user-images.githubusercontent.com/21315076/102787164-a0e8b180-436e-11eb-9d10-184046f320d4.png)